### PR TITLE
Periodic update check

### LINKED ISSUE #65: Tauri updater feature - to handle version updates distribution.

### PR DESCRIPTION
**Pull request by isSerge**
- Check for new updates daily and display the dialog with update version and release notes
- Add 'Install update' menu item (shows same dialog on click)
image
image

---
closes #65

### CODE REVIEW

File: src/lib/updater.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import * as tauri from '@tauri-apps/api';
+
+/**
+ * Emit and listen for update events, show dialog and update store if there is an update
+ * @param tauri 
+ * @param handleStoreUpdate - store method to update state in order to show 'Install update' menu item
+ * @param errorLogger 
+ */
```
Comment (achiurizo) [inline src/lib/updater.ts:8]: ✨

File: src-tauri/tauri.conf.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
       "macOS": {
         "frameworks": [],
         "minimumSystemVersion": "",
-        "exceptionDomain": "",
+        "exceptionDomain": "http://localhost:3009",
```
Comment (achiurizo) [inline src-tauri/tauri.conf.json:36]: Is this for dev?

**Review by achiurizo — APPROVED** [review]

File: src-tauri/tauri.conf.json (hunk 2 above)
Comment (isSerge) [reply src-tauri/tauri.conf.json:36]: yes, sometimes it complains about security policy, [docs](https://tauri.app/v1/guides/building/macos/#application-bundle-customization) recommend adding `localhost`

File: src-tauri/tauri.conf.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
       "endpoints": [
         "https://tauri-updater.subspace.network/v1/{{target}}/{{arch}}/{{current_version}}"
       ],
-      "dialog": true,
+      "dialog": false,
```
Comment (ozgunozerk) [inline src-tauri/tauri.conf.json:51]: hmm why don't we need to enable dialog's ?

**Review by ozgunozerk — APPROVED** [review]
lgtm

File: src-tauri/tauri.conf.json (hunk 3 above)
Comment (isSerge) [reply src-tauri/tauri.conf.json:51]: has to be disabled in order for events to work - https://tauri.app/v1/guides/distribution/updater#events

**Pull request merged**